substrate/frame/alliance/src/lib.rs

```diff
@@ -308,6 +308,9 @@ pub mod pallet {
 		/// The number of blocks a member must wait between giving a retirement notice and retiring.
 		/// Supposed to be greater than time required to `kick_member`.
 		type RetirementPeriod: Get<BlockNumberFor<Self>>;
+
+		/// Provider for the block number. Normally this is the `frame_system` pallet.
+		type BlockNumberProvider: BlockNumberProvider<BlockNumber = BlockNumberFor<Self>>;
```

Comment:
Same as some other PR. I think this is fixing the BlockNumber to be the system block number. But there is no need for this.

The block number could be the one of the relay chain, in which case it doesn't need to be `= BlockNumberFor<Self>`.